pilkit/processors/utils.py

```diff
@@ -1,18 +1,25 @@
-import math
-from ..lib import Image
+from ..lib import Image, ImageMode
 
 
-def histogram_entropy(im):
+def color_count(image):
+    """ Return the number of color values in the input image --
+        this is the number of pixels times the band count
+        of the image.
     """
-    Calculate the entropy of an images' histogram. Used for "smart cropping" in easy-thumbnails;
-    see: https://raw.github.com/SmileyChris/easy-thumbnails/master/easy_thumbnails/utils.py
+    mode_descriptor = ImageMode.getmode(image.mode)
+    width, height = image.size
+    return width * height * len(mode_descriptor.bands)
 
-    """
-    if not isinstance(im, Image.Image):
-        return 0  # Fall back to a constant entropy.
-
-    histogram = im.histogram()
-    hist_ceil = float(sum(histogram))
-    histonorm = [histocol / hist_ceil for histocol in histogram]
+def histogram_entropy_py(image):
+    """ Calculate the entropy of an images' histogram. """
+    from math import log2, fsum
+    histosum = float(color_count(image))
+    histonorm = (histocol / histosum for histocol in image.histogram())
+    return -fsum(p * log2(p) for p in histonorm if p != 0.0)
 
-    return -sum([p * math.log(p, 2) for p in histonorm if p != 0])
+# Select the Pillow native histogram entropy function - if
+# available - and fall back to the Python implementation:
+histogram_entropy = hasattr(Image.Image, 'entropy') \
```

Comment:
I thing that it will be much more readable if these 3 line are replaced by:
```
histogram_entropy = getattr(Image.Image, 'entropy', histogram_entropy_py)
```
What you think? Am I interpreting it right?

Reply:
@vstoykov totally right – I think I have written this exact snippet this way elsewhere myself. Just pushed a new version. Thanks!